tools/downloader/converter.py

```diff
@@ -118,13 +145,25 @@ def main():
         requested_precisions = set(args.precisions.split(','))
         unknown_precisions = requested_precisions - common.KNOWN_PRECISIONS
         if unknown_precisions:
+            try:
+                telemetry.send_event('md', 'converter_error', str({'error': 'Unknown precisions specified.', 'model': 'none'}))
+                telemetry.end_session()
+                telemetry.force_shutdown(1.0)
+            except NameError:
+                pass
             sys.exit('Unknown precisions specified: {}.'.format(', '.join(sorted(unknown_precisions))))
 
     models = common.load_models_from_args(parser, args)
 
     output_dir = args.download_dir if args.output_dir is None else args.output_dir
 
     def convert(reporter, model):
+        try:
+            telemetry = tm.Telemetry()
+            telemetry.send_event('md', 'converter_model_name', model.name)
+            telemetry.send_event('md', 'converter_framework', model.framework)
```

Comment:
@AnastasiaKazantaeva if we send single event with dict {'model': model_name, 'framework': model.framework}, is it guarantee the same result as these 2 separated events? (I mean can we agregate statistics per model and per framework in the same time or only for union)?

Reply:
sending 2 events we can separate information: most popular models, most popular framework. If we want to know most popular models and its framework, need to send it in dict in one event. It depends on info you want to get finally

Reply:
Why do we need to send the framework at all? It's uniquely derivable from the model name.

Reply:
@IRDonch if it is included in model name, then there is no need to send framework name

Reply:
@IRDonch we don't have framework in every model name

Reply:
but we actually know which framework each model belong

Reply:
So, we don't need to aggregate statistics over frameworks? Models statistics is enough?

Reply:
@eaidova @jkamelin but do you need to quickly build info about framework? top1 - framework1, top2 - ..., top3 ...
building this stats will be complicated knowing only model name. But this stats will help to prioritize direct support about frameworks. I'd love to have both

Reply:
@AnastasiaKazantaeva my initial question was just difference between dict and string parameters sending result. I read readme in telemetry repo and this topic looks uncovered for me (now, after your answer,  I understand difference, thanks). I have nothing against frameworks :)

Reply:
I'd like to have framework statistics. @IRDonch how do you suggest to derive frameworks from models?

Reply:
@eaidova great

Reply:
> how do you suggest to derive frameworks from model names?

I don't know much about Google Analytics, but it seems that it allows you to [import custom data](https://support.google.com/analytics/topic/6064627). We could import the model-name-to-framework mapping for each release and use it in conjunction with the events to build reports involving frameworks. The mapping itself is trivial to obtain:

```sh
./tools/downloader/info_dumper.py --all | jq '.[] | [.name, .framework]'
```

The approach that you currently have implemented doesn't scale. You can get info about which frameworks are requested the most, but what if you wanted to know which frameworks cause the most failures? What if you want to group the models by task type instead? Are you going to add all model information to all events that are specific to a model?